source/UOXJSMethods.cpp

```diff
@@ -3611,20 +3613,26 @@ JSBool CChar_ResourceCount( JSContext *cx, JSObject *obj, uintN argc, jsval *arg
 	}
 	if( argc >= 3 )
 	{
-		moreVal = static_cast<UI32>( JSVAL_TO_INT( argv[2] ));
+		moreVal = static_cast<SI64>( JSVAL_TO_INT( argv[2] ));
+	}
+	if( argc >= 4 )
+	{
+		sectionId = JS_GetStringBytes( JS_ValueToString( cx, argv[3] ));
 	}
 
 	bool colorCheck = ( itemColour != -1 ? true : false );
+	bool moreCheck = ( moreVal != -1 ? true : false );
 
-	*rval = INT_TO_JSVAL( GetItemAmount( myChar, realId, static_cast<UI16>( itemColour ), moreVal, colorCheck ));
+	*rval = INT_TO_JSVAL( GetItemAmount( myChar, realId, static_cast<UI16>( itemColour ), static_cast<UI32>( moreVal ), colorCheck, moreCheck, sectionId ));
```

Comment:
If you return the type, you can lose this static_cast

Reply:
Refer to above conversation